Bottom: 3ff35ffa512f2f5ae073ba165c320f98940e2a8d
Top:    f6323ae952841a13081c291aadbfe93eb3986d48
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-01 01:56:58 +0100

hid/gtk: Attempt to fix logic to flip component / solder group visibility on flip

Should get back to better behaviour. I probably broke this somewhat with
commit f903b4be6b85efc110852f7be40edf8245f0a513, which attempted to
re-state the previous logic in a clearer fashon.

The logic now should:

If flipping sides, and only ONE of the solder / component layers (groups)
is visible, and that layer (group) is _active_, then swap the visibilities
of the component / solder layers (groups), and make the newly visible
layer (group) active.

There are still bugs in this code relating to the assumption that the
first entry in the layer group is the one which is being toggled. This
breaks if the first entry in the group is that corresponding to the silk
for that side of the board.


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 1c5c150..58f299a 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -1408,12 +1408,12 @@ SwapSides (int argc, char **argv, Coord x, Coord y)
   if ((active_group == comp_group   && comp_on   && !solder_on) ||
       (active_group == solder_group && solder_on && !comp_on))
     {
-      bool new_comp_vis = Settings.ShowSolderSide && active_group == comp_group;
+      bool new_solder_vis = Settings.ShowSolderSide;
 
       ChangeGroupVisibility (PCB->LayerGroups.Entries[comp_group][0],
-                             new_comp_vis, new_comp_vis);
+                             !new_solder_vis, !new_solder_vis);
       ChangeGroupVisibility (PCB->LayerGroups.Entries[solder_group][0],
-                             !new_comp_vis, !new_comp_vis);
+                             new_solder_vis, new_solder_vis);
     }
 
   return 0;
